distribution/xfer/transfer.go

```diff
@@ -307,6 +336,85 @@ func (tm *transferManager) SetConcurrency(concurrency int) {
 	tm.mu.Unlock()
 }
 
+var cacheSubdir = "resumable_downloads"
+
+// hashKey hashes the key to create the directory name because otherwise tche path
+// would be too long. This hash needs to be secure or else multiple downloads could
+// share the same cache directory and cause unexpected and potentially malicious behaviour.
+func hashKey(key string) string {
+	hash := sha256.New()
+	hash.Write([]byte(key))
+	encoded := make([]byte, hash.Size()*2)
+	hex.Encode(encoded, hash.Sum(nil))
+	return string(encoded)
+}
+
+func (tm *transferManager) cachePathForKeyHash(keyHash string) string {
+	return filepath.Join(tm.cacheRoot, keyHash)
+}
+
+func (tm *transferManager) cleanUpByKeyHash(keyHash string) error {
+	return os.RemoveAll(tm.cachePathForKeyHash(keyHash))
+}
+
+// CollectGarbage purges the cache directory of any caches for transfers
+// that are not in progress
+func (tm *transferManager) CollectGarbage() error {
```

Comment:
I'm not sure why this management of the cache directory is in the transfer manager. It seems specific to downloads, so wouldn't the download manager be a better place for it?